pkg/client/src/app/pages/applications/components/application-tags/application-tags.tsx

```diff
@@ -44,38 +44,58 @@ export const ApplicationTags: React.FC<ApplicationTagsProps> = ({
           const newTagTypes: Map<number, TagType> = new Map();
           const newTags: Map<number, Tag[]> = new Map();
 
-          const validResponses = tags.reduce((prev, current) => {
+          const tagValidResponses = tags.reduce((prev, current) => {
             if (current) {
               return [...prev, current.data];
             } else {
               return prev;
             }
           }, [] as Tag[]);
 
-          const validTagIds = validResponses.map((e) => e.id);
+          const validTagIds = tagValidResponses.map((e) => e.id);
           const newUnknownTagIds = application.tags
             ?.map((e) => Number(e))
             .filter((e) => !validTagIds.includes(e));
 
-          validResponses.forEach((e) => {
-            const tagType = e.tagType;
-            if (tagType) {
-              // Tag types
-              newTagTypes.set(tagType.id!, tagType);
-
-              // Tags
-              newTags.set(tagType.id!, [
-                ...(newTags.get(tagType.id!) || []),
-                e,
-              ]);
-            }
+          Promise.all(
+            tagValidResponses.map((tag) =>
+              getTagTypeById(tag?.tagType?.id || 0)
```

Comment:
Why not use React queries ?

Reply:
This is a bug fix for the existing code. Sure we can move to queries. But this PR aims to fix the bug with the current code.

Reply:
Ah yes of course it's already using legacy query, never mind.